nilearn/decoding/decoder.py

```diff
@@ -0,0 +1,876 @@
+        spatial smoothing to apply to the signal.
+
+    standardize: bool, optional. Default: True
+        If standardize is True, the time-series are centered and normed:
+        their variance is put to 1 in the time dimension.
+
+    target_affine: 3x3 or 4x4 matrix, optional. Default: None
+        This parameter is passed to image.resample_img. Please see the
+        related documentation for details.
+
+    target_shape: 3-tuple of int, optional. Default: None
+        This parameter is passed to image.resample_img. Please see the
+        related documentation for details.
+
+    low_pass: None or float, optional
+        This parameter is passed to signal.clean. Please see the related
+        documentation for details
+
+    high_pass: None or float, optional
+        This parameter is passed to signal.clean. Please see the related
+        documentation for details
+
+    t_r: float, optional. Default: None
+        This parameter is passed to signal.clean. Please see the related
+        documentation for details.
+
+    mask_strategy: {'background' or 'epi'}, optional. Default: 'background'
+        The strategy used to compute the mask: use 'background' if your
+        images present a clear homogeneous background, and 'epi' if they
+        are raw EPI images. Depending on this value, the mask will be
+        computed from masking.compute_background_mask or
+        masking.compute_epi_mask.
+
+    memory: instance of joblib.Memory or str
+        Used to cache the masking process.
+        By default, no caching is done. If a str is given, it is the
+        path to the caching directory.
+
+    memory_level: int, optional. Default: 0
+        Rough estimator of the amount of memory used by caching. Higher value
+        means more memory for caching.
+
+    n_jobs: int, optional. Default: 1.
+        The number of CPUs to use to do the computation. -1 means
+        'all CPUs'.
+
+    verbose: int, optional. Default: 0.
+        Verbosity level.
+
+    See Also
+    ------------
+    nilearn.decoding.DecoderRegressor: Regression strategies for Neuroimaging,
+    nilearn.decoding.Decoder: Classification strategies for Neuroimaging,
+    nilearn.decoding.SpaceNetClassifier: Graph-Net and TV-L1 priors/penalties
+    """
+
+    def __init__(self, estimator='svc', mask=None, cv=10, param_grid=None,
+                 screening_percentile=20, scoring=None, smoothing_fwhm=None,
+                 standardize=True, target_affine=None, target_shape=None,
+                 low_pass=None, high_pass=None, t_r=None,
+                 mask_strategy='background', is_classif=True, memory=None,
+                 memory_level=0, n_jobs=1, verbose=0):
+        self.estimator = estimator
+        self.mask = mask
+        self.cv = cv
+        self.param_grid = param_grid
+        self.screening_percentile = screening_percentile
+        self.scoring = scoring
+        self.is_classif = is_classif
```

Comment:
I noticed `classif` has been used at several places. If it means `classifier` or `classified` then we should just use the full name.

Reply:
I missed this. Will do a quick fix for this because I agree that it is a bit vague. Intend to rename the attribute to `is_classification`.